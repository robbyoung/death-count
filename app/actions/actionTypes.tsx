export enum ActionType {
    ADD_PLAYTHROUGH = 'Add Playthrough',
    ADD_GAME = 'Add Game',
    ADD_DEATH = 'Add Death',
    ADD_DEATH_DETAIL = 'Add Death Detail',
    COMPLETE_DEATH = 'Complete Death',
    ADD_OPTION = 'Add Option',
    ADD_OPTION_SET = 'Add Option Set',
    SELECT_GAME = 'Select Game',
    LOAD_STATE = 'Load State',
    COMPLETE_GAME = 'Complete Game',
    SELECT_PLAYTHROUGH = 'Select Playthrough',
    SELECT_OPTION_SET = 'Select Option Set',
    DELETE_PLAYTHROUGH = 'Delete Playthrough',
}
