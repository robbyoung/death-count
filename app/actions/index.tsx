export * from './actionTypes';
export * from './addPlaythrough';
export * from './addDeath';
export * from './addGame';
export * from './addDeathDetail';
export * from './completeDeath';