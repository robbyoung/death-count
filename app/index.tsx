import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import { Screens } from './screens';
import Home from './containers/home';
import NewDeath from './containers/newDeath';
import GamesList from './containers/gamesList';
import NewGame from './containers/newGame';
import Loading from './containers/loading';
import Startup from './containers/startup';
import PlaythroughsList from './containers/playthroughsList';

const MainNavigator = createStackNavigator(
	{
		Loading: { screen: Loading },
		Home: { screen: Home },
		NewDeath: { screen: NewDeath },
		GamesList: { screen: GamesList },
		NewGame: { screen: NewGame },
		Startup: { screen: Startup },
		PlaythroughsList: { screen: PlaythroughsList }
	},
	{
		initialRouteName: Screens.Loading,
	},
);

export default createAppContainer(MainNavigator);