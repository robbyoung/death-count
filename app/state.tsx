export interface Game {
    id: string;
    name: string;
    options: string;
}

export interface DeathOptions {
    name: string;
    options: string[];
}

export interface Playthrough {
    id: string;
    name: string;
    description: string;
    gameId: string;
}

export interface Death {
    id: string;
    playthroughId: string;
}

export interface State {
    games: Game[];
    playthroughs: Playthrough[];
    deaths: Death[];
    currentPlaythrough: string;
    currentGame: string;
}