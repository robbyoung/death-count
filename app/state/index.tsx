export * from './state';
export * from './testState';