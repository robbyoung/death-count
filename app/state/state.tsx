export interface Game {
    id: string;
    name: string;
    selected: boolean;
    complete: boolean;
}

export interface Playthrough {
    id: string;
    name: string;
    gameId: string;
    selected: boolean;
}

export interface Death {
    id: string;
    playthroughId: string;
    complete: boolean;
    details: { [id: string] : string };
}

export interface OptionSet {
    id: string;
    gameId: string;
    title: string;
    options: string[];
    selected: boolean;
}

export interface State {
    games: Game[];
    playthroughs: Playthrough[];
    deaths: Death[];
    optionSets: OptionSet[];
}