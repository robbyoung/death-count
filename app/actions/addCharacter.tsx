import { Death, Character } from '../state';
import { Action } from 'redux';

export interface AddCharacterAction extends Action {
    newCharacter: Character;
}

export function addCharacter(state: Character[], action: AddCharacterAction): Character[] {
    return [
        ...state,
        action.newCharacter,
    ]
}