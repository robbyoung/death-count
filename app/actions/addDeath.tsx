import { Death } from '../state';
import { Action } from 'redux';
import { ActionType } from './actionTypes';

export class AddDeath implements Action {
    type: ActionType = ActionType.ADD_DEATH;

    constructor(public newDeath: Death) {}
}

export function addDeath(state: Death[], action: AddDeath): Death[] {
    return [
        ...state,
        action.newDeath,
    ]
}