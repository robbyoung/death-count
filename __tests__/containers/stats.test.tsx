import React from 'react';
import renderer from 'react-test-renderer';
import store from '../../app/store';
import { loadStateAction } from '../../app/actions';
import Stats from '../../app/containers/stats';
import { createTestState } from '../../app/state';

const fakeNavigation = {
    dispatch: () => undefined,
}

describe('Stats Container', () => {
    it('Renders with no option set selected', () => {
        const state = createTestState(2, 5, 20, 0, 0);
        store.dispatch(loadStateAction(state));
        
        const component = renderer.create(<Stats navigation={fakeNavigation as any} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Renders with chart and stats', () => {
        const state = createTestState(2, 5, 20, 0, 0, 0);
        store.dispatch(loadStateAction(state));
        
        const component = renderer.create(<Stats navigation={fakeNavigation as any} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
