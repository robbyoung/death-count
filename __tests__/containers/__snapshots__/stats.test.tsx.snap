// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Container Renders with chart and stats 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#592a29",
      "height": "100%",
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "marginLeft": "5%",
            "marginRight": "5%",
            "width": "90%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 18,
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
        >
          Pick a Death Detail
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 5,
              "width": "100%",
            }
          }
        >
          <View>
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "Option Set 0",
                    "textColor": undefined,
                    "value": "o0",
                  },
                  Object {
                    "label": "Option Set 1",
                    "textColor": undefined,
                    "value": "o2",
                  },
                  Object {
                    "label": "Option Set 2",
                    "textColor": undefined,
                    "value": "o4",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "margin": 20,
          }
        }
      >
        <ARTSurfaceView
          style={
            Array [
              undefined,
              Object {
                "height": 710,
                "width": 710,
              },
            ]
          }
        >
          <ARTGroup
            opacity={1}
            transform={
              Array [
                6.123233995736766e-17,
                -1,
                1,
                6.123233995736766e-17,
                0,
                710,
              ]
            }
          >
            <ARTShape
              d={
                Array [
                  0,
                  532.5,
                  355,
                  4,
                  355,
                  355,
                  177.5,
                  0,
                  360,
                  1,
                ]
              }
              fill={null}
              opacity={1}
              stroke={
                Array [
                  1,
                  1,
                  1,
                  1,
                ]
              }
              strokeCap={1}
              strokeDash={null}
              strokeJoin={1}
              strokeWidth={355}
              transform={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            />
            <ARTShape
              d={
                Array [
                  0,
                  532.5,
                  355,
                  4,
                  355,
                  355,
                  177.5,
                  0,
                  6.283185307179586,
                  1,
                ]
              }
              fill={null}
              opacity={1}
              stroke={
                Array [
                  1,
                  1,
                  1,
                  1,
                ]
              }
              strokeCap={1}
              strokeDash={null}
              strokeJoin={1}
              strokeWidth={355}
              transform={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            />
            <ARTShape
              d={
                Array [
                  0,
                  532.5,
                  355,
                  4,
                  355,
                  355,
                  177.5,
                  0,
                  3.141592653589793,
                  1,
                ]
              }
              fill={null}
              opacity={1}
              stroke={
                Array [
                  0.3215686274509804,
                  0.8431372549019608,
                  0.14901960784313725,
                  1,
                ]
              }
              strokeCap={0}
              strokeDash={null}
              strokeJoin={1}
              strokeWidth={355}
              transform={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            />
            <ARTShape
              d={
                Array [
                  0,
                  177.5,
                  355,
                  4,
                  355,
                  355,
                  177.5,
                  3.141592653589793,
                  4.71238898038469,
                  1,
                ]
              }
              fill={null}
              opacity={1}
              stroke={
                Array [
                  1,
                  0.45098039215686275,
                  0,
                  1,
                ]
              }
              strokeCap={0}
              strokeDash={null}
              strokeJoin={1}
              strokeWidth={355}
              transform={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            />
            <ARTShape
              d={
                Array [
                  0,
                  354.99999999999994,
                  177.5,
                  4,
                  355,
                  355,
                  177.5,
                  4.71238898038469,
                  6.283185307179586,
                  1,
                ]
              }
              fill={null}
              opacity={1}
              stroke={
                Array [
                  1,
                  0,
                  0,
                  1,
                ]
              }
              strokeCap={0}
              strokeDash={null}
              strokeJoin={1}
              strokeWidth={355}
              transform={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            />
          </ARTGroup>
        </ARTSurfaceView>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#e84641",
            "borderRadius": 5,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          Option 1
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "width": "50%",
              }
            }
          >
            2
             Deaths
          </Text>
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "textAlign": "right",
                "width": "50%",
              }
            }
          >
            50
            % of Deaths
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#e84641",
            "borderRadius": 5,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          Option 2
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "width": "50%",
              }
            }
          >
            1
             Deaths
          </Text>
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "textAlign": "right",
                "width": "50%",
              }
            }
          >
            25
            % of Deaths
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#e84641",
            "borderRadius": 5,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          Option 3
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "width": "50%",
              }
            }
          >
            1
             Deaths
          </Text>
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "textAlign": "right",
                "width": "50%",
              }
            }
          >
            25
            % of Deaths
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Stats Container Renders with no option set selected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#592a29",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": "5%",
        "marginRight": "5%",
        "width": "90%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      Pick a Death Detail
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 5,
          "width": "100%",
        }
      }
    >
      <View>
        <RCTPicker
          items={
            Array [
              Object {
                "label": "None Selected",
                "textColor": undefined,
                "value": undefined,
              },
              Object {
                "label": "Option Set 0",
                "textColor": undefined,
                "value": "o0",
              },
              Object {
                "label": "Option Set 1",
                "textColor": undefined,
                "value": "o2",
              },
              Object {
                "label": "Option Set 2",
                "textColor": undefined,
                "value": "o4",
              },
            ]
          }
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
